### ISSUE TITLE
Support for Xen hypervisor domains, such as Qubes OS's dom0 for data input

### ISSUE THREAD

**Issue by Atrate** [nicolargo/glances#2184]
**Is your feature request related to a problem? Please describe.**
[Qubes OS](https://www.qubes-os.org/) is a security-focused Xen distribution that allows for the creation of (mostly Linux) virtual machines and for substantial integration between them. [Xen utilizes dom0 (domain 0) as the administrative hypervisor domain](https://wiki.xenproject.org/wiki/Dom0) (usually Linux-based) and provides the [xentop](https://wiki.xenproject.org/wiki/Xentop(1)) utility for performance monitoring. Needless to say, `xentop` is not the prettiest or most readable of software, especially compared to `glances`. If `glances` is ran on `dom0`, it treats it as a "normal" Linux distribution and only reports stats available to the `dom0` VM itself, not stats for the whole system, which **are available** to `dom0` exclusively.

**Describe the solution you'd like**
It would be nice if `glances` detected when running on a Xen `dom0` hypervisor domain and displayed stats for the whole host, all other running domains included (maybe in a similar format to Docker?), not just `dom0` itself.

**Comment by nicolargo** [comment]
Resources for dev:

- https://stackoverflow.com/questions/12187896/listing-xencenter-inventory-with-python-and-the-xenapi-module
- https://imil.net/stuff/xapi.py
- https://xapi-project.github.io/xen-api/

**Comment by Atrate** [comment]
Bump

**Comment by Atrate** [comment]
Bump

**Comment by Atrate** [comment]
Bump

**Comment by Atrate** [comment]
Bump

**Comment by nicolargo** [comment]
Glances now include a plugin called VMs: https://glances.readthedocs.io/en/develop/aoa/vms.html

A first solution is to use it "as it": This plugins support the following virtual machine (for the moment): Multipass and Virsh. 
Under the wood, Virsh uses Libvirt and Libvirt supports Xen hypervisor: https://wiki.xenproject.org/wiki/Libvirt Perhaps a first work-around is to configure Virsh to monitor Xen VM and check if Glances can display it in the VMs plugin (warning: this plugin is disabled by default, so you should enable it from the Glances configuration file or from the command line).

@Atrate Can you test it ?

Another solution is to develop a new "native" Xen connector additionally to Multipass and Virsh. Contributor are welcome.

**Comment by Atrate** [comment]
> This plugins support the following virtual machine (for the moment): Multipass and Virsh

If it supports virsh it should be able to query at least some parameters on a Qubes OS system, which is what I'm after. I presume it's going to be released in v4.3.2? Also, if I were to compile from source, what commit would you recommend to check out?

**Comment by nicolargo** [comment]
@Atrate Yes it will be release in Glances 4.3.2. But you can already test it on the develop branch.

The current develop branch is published to the test.pypi.org package index. If you want to test the develop version (could be instable), enter:

    pip install --user -i https://test.pypi.org/simple/ Glances
